### ISSUE TITLE
copy is showing in user's language regardless of URL structure

### ISSUE THREAD

**Issue by mikefantini** [DoSomethingArchive/legacy-website#5162]
if you go to the direct path https://thor.dosomething.org/br/voluntario/notas-de-shawn-pt the content should be in BR PT regardless of the user's language settings.

However, what's happening currently is drupal is determining what content to display to the user based on the user's language settings.

Language settings should really only determine what URL to take the user to, not what copy to display once on that URL.

i.e. if my language is BR PT, and I go to thor.dosomething.org/campaigns/notes-shawn, then I should instead be taken to thor.dosomething.org/br/voluntario/notas-de-shawn-pt and the content should be in BR PT.

but if I go straight to thor.dosomething.org/us/campaigns/notes-shawn, then the content should be in US EN even though my language is BR PT.

once we have the language picker, we'll allow users to specifically indicate that they want the global version, but until then I don't think we'll be allowing users to specifically access the GL EN version if a translation exists in their language.

since this is all such an edge case experience, I think this should be low priority.

see screenshots in comments below.

---

previous description:

created a Global English translation on https://www.dosomething.org/node/32/translate

a new Problem fact was referenced and different copy was written in the Supporting Statement on the Solution.

However, when you go to view https://www.dosomething.org/campaigns/13-gallon-challenge?language=en-global (which you can get to by clicking on the title in the Global English row of http://staging.beta.dosomething.org/node/1656/translate) it is not displaying the new Problem fact or the copy that is in the Supporting Statement on the Solution.

(previously the Supporting Statement on the Solution was showing the proper copy, but the Problem fact was wrong, now both are incorrect).

Everything is being stored properly in the database.

**Comment by angaither** [comment]
@mikefantini is this one still an issue?

**Comment by mikefantini** [comment]
I believe so.

When I go to https://thor.dosomething.org/br/voluntario/notas-de-shawn-pt logged in as an administrator with language set to US English I see this, even though the BR-PT fields are filled in with PT copy:

![image](https://cloud.githubusercontent.com/assets/6330971/10172143/d4d39d0c-66ab-11e5-8d80-e6119f409bf5.png)

**Comment by mikefantini** [comment]
hmmm...

changing my language to BR PT causes the proper copy to display:

![image](https://cloud.githubusercontent.com/assets/6330971/10172241/5e46c514-66ac-11e5-953b-708708cb53b7.png)

**Comment by mikefantini** [comment]
and as an auth user with language set to spanish I'm seeing spanish copy even though i'm on /br/voluntario/notas-de-shawn-pt:

![image](https://cloud.githubusercontent.com/assets/6330971/10172448/641a5f54-66ad-11e5-8bd5-dee6d7c5309e.png)

updating the title of this issue now

**Comment by mikefantini** [comment]
also as an anonymous user https://thor.dosomething.org/br/voluntario/notas-de-shawn-pt display BR PT which is correct:

![image](https://cloud.githubusercontent.com/assets/6330971/10177362/caac5b90-66c7-11e5-868a-0d748d74c72d.png)

**Comment by mikefantini** [comment]
so some weird behavior, title is the only thing that _isn't_ displaying the proper language for either auth users with MX SP as their language or anon users on https://thor.dosomething.org/mx/voluntario/gtfo-mx

auth user MX SP lang:
![image](https://cloud.githubusercontent.com/assets/6330971/10177580/d7139bae-66c8-11e5-9db5-8c32ed8ed485.png)

anon user:
![image](https://cloud.githubusercontent.com/assets/6330971/10177590/e08164aa-66c8-11e5-9e4a-8ba025f272ef.png)

however the MX SP title is translated for admin user (US EN as lang) but all other content shows in US EN:

![image](https://cloud.githubusercontent.com/assets/6330971/10177625/0fd9497a-66c9-11e5-9eb8-cf7223a9c21e.png)

and then when I go to https://thor.dosomething.org/mx/node/5459/pitch as an admin with US EN as lang, only the title is translated, everything else is in US EN:

![image](https://cloud.githubusercontent.com/assets/6330971/10177736/b5106004-66c9-11e5-8749-ab94ab0ef241.png)

and then MX SP title and all other content is translated for admin user (MX SP as lang); and everything is in MX SP on /node/5459/pitch for admin with MX SP as lang):

![image](https://cloud.githubusercontent.com/assets/6330971/10177659/399b382c-66c9-11e5-94f1-c872860e501a.png)

so basically title is acting very bizarre specifically for GTFO on the MX translation (which if memory serves me we updated after saving that translation, so maybe that explains why).

**Comment by mshmsh5000** [comment]
Switching the interface language detection settings to favor "URL" over "User" seems to fix all of this behavior.

![interface-lang](https://cloud.githubusercontent.com/assets/149811/10232750/6d38d2d0-6858-11e5-976a-993dafbd5e59.png)

**Comment by mshmsh5000** [comment]
I would like to congratulate @mikefantini for the number of screenshots in this issue.

**Comment by mikefantini** [comment]
everything is working properly except for GTFO is still being whack. I'll check this campaign out again once we update the thor db with production data
